### ISSUE TITLE
Import-errors on lines that don't exist at the bottom of the file instead of the top?

### ISSUE THREAD

**Issue by djch** [AtomLinter/linter-flake8#211]
![l618](https://cloud.githubusercontent.com/assets/4392286/16361822/7b7eadd2-3bdf-11e6-96ed-d480ce159edb.png)

Pictured is the last line of the file, L617. Note that it's calling out an import error from the top of the file on L618 which does not exist.

I could swear this started when I was briefly experimenting with an `isort` atom plug-in which I've since uninstalled.

Any ideas?

**Comment by Arcanemagus** [comment]
Hmmm, the line number should be coming straight from `flake8`, what do you see when you run it manually on the CLI?

**Comment by djch** [comment]
I've just run `flake8` over a few files and it seems to show the correct line numbers for the import problems.

I s'pose the other change that happened just recently was flake8 2.6.x was released.

**Comment by Arcanemagus** [comment]
Can you do the following:

* Find a file that is reporting an improper line in Atom through `linter-flake8`
* Save the file in the state that the error is showing
* Open up your CLI and run `flake8` _on that file_
* Check the line number in the `flake8` output

**Comment by djch** [comment]
I just noticed that it's not happening on *all* files that unused import problems; I just added an unused import for the sake of a screenshot but it was fine...

Anyway, as asked, here's an example of a file that does have the problem. Thanks for looking into it.

![atom-linter-top](https://cloud.githubusercontent.com/assets/4392286/16479909/9c285d6c-3ee6-11e6-9e47-8383880c75f3.png)
![atom-linter](https://cloud.githubusercontent.com/assets/4392286/16479911/9c897890-3ee6-11e6-9af1-d48beaf7b09e.png)
![cli](https://cloud.githubusercontent.com/assets/4392286/16479910/9c8466e8-3ee6-11e6-8ba1-9ae38857e9e9.png)

**Comment by Arcanemagus** [comment]
Hmmm, can you open Atom's Developer Tools, place a breakpoint [here](https://github.com/AtomLinter/linter-flake8/blob/v1.13.2/lib/main.coffee#L228), and see what `output` contains?

**Comment by djch** [comment]
Sorry I'm not a JS/coffee person so I don't know what you mean by adding a breakpoint. I can get as far as opening the dev tools, I could follow some instructions from there?

Also, sharing a file won't help because the same thing isn't happening for my colleagues working on the same project, with the plugin. Must be related to my specific environment somehow.

I've also figured out that it isn't affecting some files and not others but that only `from` imports don't work correctly. Plain unused `import` lines _are_ highlighted properly.

**Comment by Arcanemagus** [comment]
After opening Developer Tools you would need to click on the "Source" tab at the top, then find the `lib/main.coffee` file in the tree view on the left. Once you have the file open scroll to the line number I linked to above, and click the line number. It should be highlighted with a blue "tag". Next trigger a lint and the execution should stop at that line. You can then see what `output` contains in the "Scope" section on the right, or by simply typing `output` in the console.

**Comment by hvdklauw** [comment]
Ran into the same problem.

Code:

```python
from fabric.api import task, local, lcd

from . import api, bo, fo
from ..helpers import LayerTask


@task(default=True, task_class=LayerTask)
def full_deploy(tag=None):
    """Deploy FO, BO and API."""
    print("Deploying all")
    api.deploy(tag=tag)
    fo.deploy(tag=tag)
    bo.deploy(tag=tag)

```


There is not `output` variable, but here is `result.stdout`:

```
stdin:1:1: F401 'fabric.api.lcd' imported but unused stdin:1:1: F401 'fabric.api.local' imported but unused
```



output of `apm list | grep flake8`

```
├── linter-flake8@1.13.2
```

and ```flake8 --version```
```
2.6.2 (pycodestyle: 2.0.0, pyflakes: 1.2.3, mccabe: 0.5.0) CPython 3.5.1 on Darwin
```

**Comment by hvdklauw** [comment]
hmm the input to `extractRange` is correct. So it must be somewhere in the handling of the F401 error in there...

**Comment by hvdklauw** [comment]
I was wondering why this all looked so familiar... 2eb8a3f91299b9cc04e06fe4c53eb52f9217a6dc

**Comment by hvdklauw** [comment]
OK, the problem is [here](https://github.com/AtomLinter/linter-flake8/blob/2eb8a3f91299b9cc04e06fe4c53eb52f9217a6dc/lib/main.coffee#L55)

The symbol is `fabric.api.lcd` while the token value is just `lcd`

I'm guessing the error in flake8 changed and now gives the full python path instead of just the symbol.

**Comment by dikamilo** [comment]
This change resolves problem, I hope it will be released soon ;)
